pkg/storage/tsdb/bucketindex/updater.go

```diff
@@ -123,7 +123,7 @@ func (w *Updater) updateBlockIndexEntry(ctx context.Context, id ulid.ULID) (*Blo
 	metaFile := path.Join(id.String(), block.MetaFilename)
 
 	// Get the block's meta.json file.
-	r, err := w.bkt.Get(ctx, metaFile)
+	r, err := w.bkt.ReaderWithExpectedErrs(w.bkt.IsObjNotFoundErr).Get(ctx, metaFile)
```

Comment:
Should we have a test on this?

Reply:
Let me see if it is easy to setup `metricBucket` in the test to make sure the metric not increased in this case.

Reply:
Added test